ui v2: restyle resize workflows

scarlettperry opened the pull request:
### Description
PR
* adds style updates made in https://github.com/lyft/clutch/pull/781 to the resize workflows (EC2 & K8S)
* updates MutableRow to have disabled text field and icon
* updates EC2 term instance confirmation page to use `useDataLayout.displayValue`
* adds MutableRow story

Resize ASG
![asg](https://user-images.githubusercontent.com/39421794/102399520-bcd70680-3fae-11eb-8621-fefc59f88f90.gif)

Resize HPA
![hpa](https://user-images.githubusercontent.com/39421794/102399789-16d7cc00-3faf-11eb-8dc5-95beed46e23d.gif)

<img width="400" alt="Screen Shot 2020-12-16 at 2 49 26 PM" src="https://user-images.githubusercontent.com/39421794/102398923-e9d6e980-3fad-11eb-9fdb-1861ceb38fde.png">

### Testing Performed
Locally

scarlettperry commented on frontend/workflows/ec2/src/resize-asg.tsx:
```diff
@@ -87,26 +87,24 @@ const GroupDetails: React.FC<WizardChild> = () => {
+          { name: "New Max Size", value: group.size.max },
+          { name: "New Desired Size", value: group.size.desired },
+        ]}
+      />
```
didn't add the accordion since we don't have metadata like annotations/labels for an [ASG](https://github.com/lyft/clutch/blob/main/api/aws/ec2/v1/ec2.proto#L67)

scarlettperry commented on frontend/packages/core/src/Table/metadata-table.tsx:
```diff
@@ -65,7 +67,7 @@ const TableCell = styled(MuiTableCell)({
   fontSize: "14px",
   fontWeight: "normal",
   height: "48px",
-  padding: "0 16px",
```
on figma, the table cells have a padding top/bottom 15px, left/right 16px

scarlettperry commented on frontend/packages/core/src/Table/metadata-table.tsx:
```diff
@@ -65,7 +67,7 @@ const TableCell = styled(MuiTableCell)({
   fontWeight: "normal",
   height: "48px",
-  padding: "0 16px",
+  padding: "15px 16px 15px 16px",
```
based on figma design for table cell

dschaller replied:
This one is hard because it depends on which cell you are looking at. Some of them have top & bottom padding but others don't. I think for now we shouldn't have any and just center the items vertically

scarlettperry replied:
okay sounds good. i reverted to what we currently have (`padding: "0 16px"`) bc didn't look good w/o some right padding 

ex w/o the right padding
<img width="600" alt="Screen Shot 2020-12-16 at 5 52 15 PM" src="https://user-images.githubusercontent.com/39421794/102421064-82329580-3fd1-11eb-8c68-f3664e6ff795.png">

scarlettperry commented on frontend/packages/core/src/Table/metadata-table.tsx:
```diff
@@ -114,18 +116,22 @@ const MutableRow: React.FC<MutableRowProps> = ({ data, onUpdate, onReturn, valid
-          error={!!error || false}
-        />
+        <Grid container alignItems="center" wrap="nowrap" spacing={2}>
+          <TextField disabled id={data.id} name={data.name} defaultValue={data.value} />
```
the text field widths are larger than figma design. i can override by giving them fix widths but unsure if that's best practice

dschaller replied:
Let's override these for now to match the designs

dschaller commented on frontend/workflows/ec2/src/resize-asg.tsx:
```diff
@@ -68,10 +66,12 @@ const GroupDetails: React.FC<WizardChild> = () => {
         ]}
       />
       <ButtonGroup
+        justify="flex-end"
```
ah I just landed a PR to change the button group interface. You'll probably need to pick that up :)

dschaller commented on frontend/workflows/ec2/src/terminate-instance.tsx:
```diff
@@ -79,15 +79,18 @@ const InstanceDetails: React.FC<WizardChild> = () => {
+
+  const data = [
+    { name: "Instance ID", value: instance.instanceId },
+    { name: "Region", value: instance.region },
```
Thank you!

dschaller commented on frontend/workflows/k8s/src/resize-hpa.tsx:
```diff
@@ -65,11 +80,18 @@ const HPADetails: React.FC<WizardChild> = () => {
           { name: "Cluster", value: hpa.cluster },
         ]}
       />
+      <Accordion title="Metadata">
```
Probably want to wrap this to only display it if we have metadata

dschaller commented on frontend/workflows/ec2/src/resize-asg.tsx:
```diff
@@ -87,26 +87,24 @@ const GroupDetails: React.FC<WizardChild> = () => {
+      <Confirmation action="Resize">{notes && formattedNotes}</Confirmation>
+      <MetadataTable
+        data={[
+          { name: "Name", value: group.name },
```
Don't worry too much about this as part of this PR but I know @danielhochman mentioned adding the previous values into this screen at some point with Kanchan. We should probably follow up about that to make sure we are ok not showing those.

scarlettperry replied:
ah okay sounds good! will follow up
I'll also leave a to-do for a reminder

scarlettperry commented:
addressed the feedback. this is how the mutable rows look now. for the disabled text field, I also had to remove the default padding b/c at the smaller width, you couldn't see the input value.

<img width="854" alt="Screen Shot 2020-12-16 at 7 07 52 PM" src="https://user-images.githubusercontent.com/39421794/102421284-1a307f00-3fd2-11eb-974f-8d32cdd089d3.png">

danielhochman commented on frontend/workflows/k8s/src/resize-hpa.tsx:
```diff
@@ -40,6 +41,20 @@ const HPADetails: React.FC<WizardChild> = () => {
+  const metadata = [];
+
+  if (hpa.annotations) {
+    Object.keys(hpa.annotations).forEach(key => {
```
nit: recommend lodash forEach

scarlettperry replied:
@danielhochman , jst want to triple check this is what you had in mind https://github.com/lyft/clutch/pull/794/commits/6c86a0a9f48c588c9d4a20a05a790fc292603443.

could have also done `_.keys(something).forEach(key => {`

danielhochman replied:
yep looks good!

danielhochman reviewed (approved):
lgtm, one nit
